agent/lib/src/utils.dart

```diff
@@ -649,3 +649,20 @@ Future<Null> unlockKeyChain() async {
         canFail: false, silent: true);
   }
 }
+
+Future<Null> cpFileToGcs(String file, String gsBucket) async {
+  if (Platform.isLinux || Platform.isMacOS) {
+    await eval('gsutil.py', ['cp', file, gsBucket],
```

Comment:
where is gsutil.py coming from? is this depot_tools gsutil?

Reply:
I took this code from here: https://github.com/flutter/cocoon/compare/master...keyonghan:save_log_to_gcs

Reply:
I tried https://cloud.google.com/storage/docs/gsutil_install#mac, and it worked on agent bots.
To make agents work, gsutil needs to be installed beforehand, and also we need to copy over the authentication (key & .boto).

Reply:
Are those installation and authentication steps already part of creating a new agent? What work is needed for that outside of this PR?

Reply:
(Sorry missed this comment!)
No, they are not part of the agent at this moment. We need to install/authenticate with extra work.

I would suggest test on a specific agent first to make sure CI works as expected before we submit this change and synchronize across all agents. Also, it would be good to compare test execution time before and after this change.

> What work is needed for that outside of this PR?
Install the gsutil; generate `.boto` file and copy over to the agent, also the key json file; (i remembered I have already installed on some Mac and Linux bots), etc.

Please check the commands/steps I sent earlier, and let me know if you need help access the devicelab agents.